crates/ruff_cli/tests/integration_test.rs

```diff
@@ -27,315 +28,166 @@ use ruff_cli::args::Args;
 use ruff_cli::run;
 
 const BIN_NAME: &str = "ruff";
+const STDIN_BASE_OPTIONS: &[&str] = &["--isolated", "--no-cache", "-", "--format", "text"];
 
 #[test]
-fn stdin_success() -> Result<()> {
-    let mut cmd = Command::cargo_bin(BIN_NAME)?;
-    cmd.args(["-", "--format", "text", "--isolated"])
-        .write_stdin("")
-        .assert()
-        .success();
-    Ok(())
+fn stdin_success() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .pass_stdin(""));
 }
 
 #[test]
-fn stdin_error() -> Result<()> {
-    let mut cmd = Command::cargo_bin(BIN_NAME)?;
-    let output = cmd
-        .args(["-", "--format", "text", "--isolated"])
-        .write_stdin("import os\n")
-        .assert()
-        .failure();
-    assert_eq!(
-        str::from_utf8(&output.get_output().stdout)?,
-        r#"-:1:8: F401 [*] `os` imported but unused
+fn stdin_error() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .pass_stdin("import os\n"), @r#"
```

Comment:
Is this an inline snapshot? I'm a huge fan of inline snapshot if the content isn't massive because I then don't need switching between test and expected output.